core/src/main/java/org/jobrunr/storage/sql/common/JobTable.java

```diff
@@ -54,8 +58,9 @@ public JobTable(Connection connection, Dialect dialect, String tablePrefix, JobM
                 .withVersion(AbstractJob::getVersion)
                 .with(FIELD_JOB_AS_JSON, jobMapper::serializeJob)
                 .with(FIELD_JOB_SIGNATURE, JobUtils::getJobSignature)
-                .with(FIELD_SCHEDULED_AT, job -> job.hasState(StateName.SCHEDULED) ? job.<ScheduledState>getJobState().getScheduledAt() : null)
-                .with(FIELD_RECURRING_JOB_ID, job -> job.getRecurringJobId().orElse(null));
+                .with(FIELD_SCHEDULED_AT, job -> job.getLastJobStateOfType(ScheduledState.class).map(ScheduledState::getScheduledAt).orElse(null))
```

Comment:
It should not be null because of `getRecurringJobsLatestScheduledRun` which depends on the `scheduledAt` column. Also keeping it makes it consistent with how MongoDB behaves (it's never set back to `null`).